File: 프로그래머스/2/42747. H－Index/H－Index.py
def solution(citations):
    citations.sort()
    min =0
    max = citations[-1]
    while min<= max:
        mid = (min+max)//2
        h= len(list(filter(lambda v : v>=mid , citations)))
        rest = len(citations) - h
        if h>=mid and rest <= mid:
            min = mid + 1
        else:
            max = mid - 1
    return max

File: 프로그래머스/2/42747. H－Index/H－Index.java
import java.util.*;
import java.util.stream.*;
import java.io.*;
class Solution {
    public int solution(int[] citations) {
        Integer[] intArray = Arrays.stream(citations).boxed().toArray(Integer[]::new);
        List<Integer> list = new ArrayList<>(Arrays.asList(intArray));
        
        Collections.sort(list);
        int max = Collections.max(list);
        int h = 0;
        
        int previousPassed = 0;
        while(h<=max){
            int passed = 0;
            int idx = 0;
            while(idx< list.size() ){
                if(h <= list.get(idx)){
                    passed++;
                }
                idx++;
            }
            
            int notpassed = list.size() - passed;
            if(passed >= h && notpassed <= h){
                h++;
                if(h > max){
                    h-=1;
                    break;
                }
            }
            else{
                h--;
                break;
            }
        }
        return h;
    }
}